wagtail_events/abstract_models.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding:utf8 -*-
+
+from __future__ import unicode_literals
+
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.db import models
+from wagtail.wagtailadmin.edit_handlers import FieldPanel
+from wagtail.wagtailcore.models import Page
+
+from wagtail_events.managers import EventInstanceManager
+from wagtail_events.utils import _DATE_FORMAT_RE
+
+
+class AbstractPaginatedIndexPage(Page):
+    """ """
+    paginate_by = models.PositiveIntegerField(blank=True, null=True)
+    content_panels = Page.content_panels + [FieldPanel('paginate_by')]
+
+    class Meta(object):
+        """Django model meta options."""
+        abstract = True
+
+    def _get_children(self, request):
```

Comment:
Since we're not filtering on the specific page model fields we could probably drop this in favour of the get_children method already available on the page instance (Unless I'm missing something of course).